### DISCUSSION TITLE
Can't get foreground_templates to work with Azure segment

### DISCUSSION THREAD

**Discussion by jarlesteinsland** [JanDeDobbeleer/oh-my-posh#1189] [Troubleshoot] [answered]
I'm trying to get the Azure segment to change foreground color based on the name of the Azure subscription that's set using `az account set --subscription "<subscription-name>"`.

I'm using oh-my-posh version 6.1.1, installed with `winget`, with Powershell version 7.1.5 in Windows Terminal version 1.11.2921.0.

The config I'm using is this:

```json
{
  "type": "az",
  "style": "powerline",
  "powerline_symbol": "\uE0B0",
  "foreground": "#000000",
  "background": "#9ec3f0",
  "foreground_templates": [
    "{{if contains \"Production\" .Name}}#F1184C{{end}}",
    "{{if contains \"Development\" .Name}}#6CA35E{{end}}"
  ],
  "properties": {
    "prefix": " \uFD03 ",
    "template": "{{.Name}}"
  }
}
```

The name of the subscription displayed in the prompt changes when I change subscriptions, but the foreground color doesn't. I'm doing something similar with the Kubectl segment, where it works as I expect, so I'm not sure what I'm doing wrong here.

Is there something I'm missing? Any help with this would be appreciated.

**Comment by JanDeDobbeleer** [comment, marked as answer]
@jarlesteinsland that's a bug. Fix incoming -> https://github.com/JanDeDobbeleer/oh-my-posh/pull/1191

**Reply by jarlesteinsland** [reply]
Great, thanks!

**Reply by spinard65** [reply]
NEVERMIND!  Should have looked at the commit.  The fix for this is Azure-specific.

I just posted this on StackOverflow:
https://stackoverflow.com/questions/70037790/cant-get-oh-my-posh-foreground-templates-to-work-for-text-segment

Is it perhaps the same issue?

**Reply by JanDeDobbeleer** [reply]
@spinard65 that's because the text segment doesn't have those template properties. You're looking for the [session](https://ohmyposh.dev/docs/session) segment.
